portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,13 +45,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
+        int currPage = Integer.parseInt(request.getParameter("page"));
+
         //get number of max comments to show from user input and limit list of comments
         int numComments = this.getMaxNumComments(request);
         List<Entity> limitedComments = results.asList(FetchOptions.Builder.withLimit(numComments));
-        
+
         //add entity to list of comments
         List<Comment> commentsList = new ArrayList<Comment>();
-        for (Entity entity : limitedComments){
+        for (int i = 5*currPage; i <= Math.min(5*currPage + 4, limitedComments.size()-1); i++){
```

Comment:
May I ask what the 5 and 4 are referring to here? Generally we want to try to avoid "magic numbers" ("arbitrary numerical constant. In other words, a random number that "just works" or "looks right" without a grounded explanation") as much as possible unless they are absolutely necessary. It looks like your drop down to change the amount of visible comments is not doing anything at the moment, could that issue possibly be related here? If so, how could we take both pagination and the max number of comments a user wants to see at one time into consideration?

Reply:
the 5 and 4 are used to calculate which comments to load onto the page. For example, if the currPage = 0 (we are on the first page of comments) the we would have for (int i = 0; i<=4;i++) and index into the arraylist of comments to pull up the correct comments to display. Similarly, if we are on currPage = 1 (second page of comments) then the for loop would be for (int i = 5; i<=9; i++). Should I store 5 and 4 as a constant instead? Also the drop down does limit the number of comments. So for example, if you limit to 5 comments, and then you click the next button, there wouldn't be any comments to show. I don't think the dropdown is very necessary in terms of user experience given that I have added pagination, but it was mentioned in the walkthrough so I left it in there

Reply:
I think I had a different understanding on the functionality of the dropdown. I was thinking that it would limit the total number of comments. However, I have no changed it so that the dropdown acts as a way to set how many comments are displayed per page.